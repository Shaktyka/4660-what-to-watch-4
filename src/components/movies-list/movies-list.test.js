import React from 'react';
import renderer from 'react-test-renderer';
import MoviesList from './movies-list.jsx';

const MOCK_FILMS = [
  {
    title: `Mindhunter`,
    preview: `mindhunter.jpg`
  },
  {
    title: `Moonrise Kingdom`,
    preview: `moonrise-kingdom.jpg`
  }
];

describe(`MoviesList rendering`, () => {

  it(`MoviesList renders correctly`, () => {

    const tree = renderer
      .create(
          <MoviesList
            films={MOCK_FILMS}
            mainTitleClickHandler={() => {}}
            onHoverCard={() => {}}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

});
