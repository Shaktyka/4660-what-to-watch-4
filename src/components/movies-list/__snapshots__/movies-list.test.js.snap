// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList rendering MoviesList renders correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Mindhunter"
        height="175"
        src="img/mindhunter.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Mindhunter
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Moonrise Kingdom"
        height="175"
        src="img/moonrise-kingdom.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Moonrise Kingdom
      </a>
    </h3>
  </article>
</div>
`;
